scripts/build/configure/termux_step_configure_cmake.sh

```diff
@@ -14,9 +14,25 @@ termux_step_configure_cmake() {
 	else
 		MAKE_PROGRAM_PATH=$(which make)
 	fi
-	CFLAGS+=" --target=$CCTERMUX_HOST_PLATFORM -fno-addrsig"
-	CXXFLAGS+=" --target=$CCTERMUX_HOST_PLATFORM -fno-addrsig"
-	LDFLAGS+=" --target=$CCTERMUX_HOST_PLATFORM"
+
+	CXXFLAGS+=" -fno-addrsig"
+	CFLAGS+=" -fno-addrsig"
+
+	local CMAKE_ADDITIONAL_ARGS=()
+	if [ -z "$TERMUX_ON_DEVICE_BUILD" ]; then
+		CXXFLAGS+=" --target=$CCTERMUX_HOST_PLATFORM"
+		CFLAGS+=" --target=$CCTERMUX_HOST_PLATFORM"
+		LDFLAGS+=" --target=$CCTERMUX_HOST_PLATFORM"
+
+		CMAKE_ADDITIONAL_ARGS+=("-DCMAKE_CROSSCOMPILING=True")
+		CMAKE_ADDITIONAL_ARGS+=("-DCMAKE_LINKER=$TERMUX_STANDALONE_TOOLCHAIN/bin/$LD $LDFLAGS")
+		CMAKE_ADDITIONAL_ARGS+=("-DCMAKE_SYSTEM_NAME=Android")
```

Comment:
Hmm, this ties into the issue I raised with #151, most packages that use this will likely require setting this on-device too.

Reply:
`-DCMAKE_SYSTEM_NAME=Android` seems require option for standalone toolchain.

Reply:
OK, someone mentioned that on that issue a couple years ago. I'm already proposing patching CMake to make sure CMake sets that variable to Android on Termux, I will look into what else is necessary, such as modifying that file you quote also.

Once that's done, this won't need to be repeated for the on-device build. I just want to know from you and @fornwall that you'd accept a CMake patch to change that, as Fredrik seemed to want it a couple years ago, before I look into submitting a pull for CMake here.

You mentioned earlier that some packages aren't building with the on-device clang, some of those could be because this CMake variable is not set on-device.